src/platform/datapath_raw.c

```diff
@@ -182,25 +182,43 @@ CxPlatSocketCreateUdp(
         goto Error;
     }
 
+    CxPlatZeroMemory(*NewSocket, sizeof(CXPLAT_SOCKET));
     CxPlatRundownInitialize(&(*NewSocket)->Rundown);
     (*NewSocket)->Datapath = Datapath;
     (*NewSocket)->CallbackContext = Config->CallbackContext;
+
     if (Config->RemoteAddress) {
+        CXPLAT_FRE_ASSERT(!QuicAddrIsWildCard(Config->RemoteAddress));  // No wildcard remote addresses allowed.
         (*NewSocket)->Connected = TRUE;
         (*NewSocket)->RemoteAddress = *Config->RemoteAddress;
-        (*NewSocket)->LocalAddress = Datapath->ClientIP;
-    } else {
-        (*NewSocket)->Connected = FALSE;
-        CxPlatZeroMemory(&(*NewSocket)->RemoteAddress, sizeof(QUIC_ADDR));
-        (*NewSocket)->LocalAddress = Datapath->ServerIP;
     }
-    if (Config->LocalAddress && Config->LocalAddress->Ipv4.sin_port != 0) {
-        (*NewSocket)->LocalAddress.Ipv4.sin_port =
-            Config->LocalAddress->Ipv4.sin_port;
+
+    if (Config->LocalAddress) {
+        if (QuicAddrIsWildCard(Config->LocalAddress)) {
```

Comment:
This code seems to support two possibilities: either a LocalAddress that's wildcard, or no LocalAddress. Are both of those codepaths currently used by the core QUIC code?

Reply:
Are you asking for just unconnected sockets or both connected and unconnected?

Reply:
I'm specifically asking here about the two copy-pasted bits where we set LocalAddress to ClientIP, so connected.

Reply:
Er, actually I mean both. I'm saying that Config->LocalAddress==NULL and QuicAddrIsWildCard(Config->LocalAddress) seem to have the same meaning, and wondering why there need to be two different ways to do the same thing.

Reply:
Here are the supported used cases for QUIC:

Listener:
Remote address always null
Local address is wildcard or null

Client:
Remote address always fully specified
Local address may be null, fully wildcard, only port specified or fully specified

Reply:
So zeroing in on the listener case, why would Config->LocalAddress be null versus fully wildcard?

I'm trying to understand why both cases exist, because having both cases makes the code here more complicated.

Reply:
We support cases where a listener didn't specify a port up front (mostly for testing).

Reply:
K